src/cocotb_tools/pytest/plugin.py

```diff
@@ -0,0 +1,666 @@
+            The entry function must be a callable matching this form: ``entry_function(argv: List[str]) -> None``
+        """,
+    ),
+)
+
+
+def options_for_documentation() -> ArgumentParser:
+    """It helps to attach plugin options into Sphinx documentation using the sphinx-argparse extension."""
+    parser: ArgumentParser = ArgumentParser(
+        prog="pytest",
+        description="Plugin options",
+    )
+
+    for option in OPTIONS:
+        default: Any
+
+        if option.extra.get("action") == "store_true":
+            default = False
+        else:
+            default = option.default_in_help or option.default
+
+        parser.add_argument(
+            option.argument,
+            help=(
+                f"{textwrap.dedent(option.description)}\n\n"
+                f"Configuration option: ``{option.name}``\n\n"
+                f"Environment variable: :envvar:`{option.environment}`"
+            ),
+            default=default,
+            **option.extra,
+        )
+
+    return parser
+
+
+@fixture(scope="session")
+def dut() -> SimHandleBase:
+    """A cocotb fixture that is providing a simulation handle to DUT.
+
+    Provided simulation handle is the same as :py:data:`cocotb.top`.
+
+    Example usage:
+
+    .. code:: python
+
+        async def test_dut_feature_1(dut) -> None:
+            dut.rst.value = 0
+            dut.clk.value = 0
+
+            # Testing DUT feature
+            ...
+
+    Returns:
+        Simulation handle to DUT (equivalent to :py:data:`cocotb.top`).
+    """
+    return cocotb.top
+
+
+@fixture
+def hdl(request: FixtureRequest) -> HDL:
+    """A cocotb fixture that is providing a helper instance to define own HDL design, built it and
+    run set of cocotb tests from test modules (testbenches) against selected top level design.
+
+    It contains own instance of :py:class:`~cocotb_tools.runner.Runner` that can be accessed directly
+    from :py:attr:`~cocotb_tools.pytest.hdl.HDL.runner` member.
+
+    Defined HDL design can be build and test by invoking the :py:meth:`~cocotb_tools.pytest.hdl.HDL.test()` method
+    from **non-async** test functions. This method will invoke build and test steps from :py:attr:`~cocotb_tools.pytest.hdl.HDL.runner` member.
+
+    Requested fixture will pass various plugin :ref:`options <pytest-plugin-options>`
+    to own instance of :py:class:`~cocotb_tools.runner.Runner`. Like setting a desired verbosity level for cocotb runner.
+
+    Please refer to available public members of :py:class:`~cocotb_tools.pytest.hdl.HDL` that can be used to define own HDL design.
+
+    Example usage:
+
+    .. code:: python
+
+        import pytest
+        from cocotb_tools.pytest.hdl import HDL
+
+
+        @pytest.fixture
+        def my_hdl_module(hdl: HDL) -> HDL:
+            hdl.toplevel = "my_hdl_module"
+
+            hdl.sources = (
+                # Add more HDL source files here
+                ...
```

Comment:
The docs try to parse these code blocks, so try to make them valid.
```suggestion
                # ...
```